src/DictList.elm

```diff
@@ -938,6 +965,174 @@ fromDict dict =
+-}
+dropWhile : ((comparable, a) -> Bool) -> DictList comparable a -> DictList comparable a
+dropWhile predicate list =
+  toList list
+    |> List.Extra.dropWhile predicate
+    |> fromList
+
+{-| Remove duplicate values, keeping the first instance of each element which appears more than once.
+
+    unique [0,1,1,0,1] == [0,1]
+-}
+unique : DictList comparable1 comparable2 -> DictList comparable1 comparable2
+unique list =
+  toList list
+    |> List.Extra.unique
+    |> fromList
+
+{-| Drop duplicates where what is considered to be a duplicate is the result of first applying the supplied function to the elements of the list.
+-}
+uniqueBy : (comparable1 -> a -> comparable2) -> DictList comparable1 a -> DictList comparable1 a
+uniqueBy f list =
+  toList list
+    |> List.Extra.uniqueBy (uncurry f)
+    |> fromList
+
+{-| Indicate if list has duplicate values.
+
+    allDifferent [0,1,1,0,1] == False
+-}
+allDifferent : DictList comparable1 comparable2 -> Bool
+allDifferent list =
+  -- @TODO Should this method just use the values, or also the keys
+  values list
+   |> List.Extra.allDifferent 
+
+{-| Indicate if list has duplicate values when supplied function are applyed on each values.
+-}
+allDifferentBy : (comparable1 -> a -> comparable2) -> DictList comparable1 a -> Bool
+allDifferentBy f list = 
+  toList list
+    |> List.Extra.allDifferentBy (uncurry f)
+
+{-| Map functions taking multiple arguments over multiple lists. Each list should be of the same length.
+
+    ((\a b c -> a + b * c)
+        |> flip map [1,2,3]
+        |> andMap [4,5,6]
+        |> andMap [2,1,1]
+    ) == [9,7,9]
+-}
+andMap : DictList comparable a -> DictList comparable (a -> b) -> DictList comparable b
+andMap l fl =
+  let
+    keyList = keys l
+    lList = values l
+    flList = values fl
+  in
+      List.Extra.andMap lList flList
+      |> List.Extra.zip keyList
+      |> fromList
+
+{-| Equivalent to `concatMap`. For example, suppose you want to have a cartesian product of [1,2] and [3,4]:
+
+    [1,2] |> andThen (\x -> [3,4]
+          |> andThen (\y -> [(x,y)]))
+
+will give back the list:
+
+    [(1,3),(1,4),(2,3),(2,4)]
+
+Now suppose we want to have a cartesian product between the first list and the second list and its doubles:
+
+    [1,2] |> andThen (\x -> [3,4]
+          |> andThen (\y -> [y,y*2]
+          |> andThen (\z -> [(x,z)])))
+
+will give back the list:
+
+    [(1,3),(1,6),(1,4),(1,8),(2,3),(2,6),(2,4),(2,8)]
+
+Advanced functional programmers will recognize this as the implementation of bind operator (>>=) for lists from the `Monad` typeclass.
+-}
+andThen : (comparable -> a -> DictList comparable b) -> DictList comparable a -> DictList comparable b
+andThen =
+    concatMap
+
+concatMap : (comparable -> a -> DictList comparable b) -> DictList comparable a -> DictList comparable b
+concatMap f xs = empty
+--  map f xs
+--    |> toList
+--    |> values
+--    |> concat
+
```

Comment:
Ah, yes, it isn't obvious to me either what `andMap` and `andThen` ought to do in the context of `DictList` -- that would take some thinking.